sphinx/ext/autodoc/directive.py

```diff
@@ -49,19 +49,25 @@ def __getitem__(self, key):
 class DocumenterBridge:
     """A parameters container for Documenters."""
 
-    def __init__(self, env, reporter, options, lineno):
-        # type: (BuildEnvironment, Reporter, Options, int) -> None
+    def __init__(self, env, reporter, options, lineno, state):
+        # type: (BuildEnvironment, Reporter, Options, int, Any) -> None
         self.env = env
         self.reporter = reporter
         self.genopt = options
         self.lineno = lineno
+        self.state = state
         self.filename_set = set()  # type: Set[str]
         self.result = StringList()
 
     def warn(self, msg):
         # type: (str) -> None
         logger.warning(msg, location=(self.env.docname, self.lineno))
 
+    @property
+    def settings(self):
```

Comment:
I feel this is not docutils' way. Directive class does not have settings object directly.